### PR TITLE
Gossipsub: ignore duplicate publish errors

### LINKED ISSUE #5140: Validator: wrong publish error log
**Describe the bug**

Found this in the log

```
validator-2023-02-09.log:39664:Feb-09 09:04:14.522[]                error: Error proposing block slot=5759119, validator=0x810e…7963  - Internal Server Error: PublishError.Duplicate - Failed to publish block
validator-2023-02-10.log:8356:Feb-10 01:56:27.419[]                error: Error proposing block slot=5764180, validator=0xa8dc…7536  - Internal Server Error: PublishError.Duplicate - Failed to publish block
```

However as we can see those 2 blocks are all success

**Expected behavior**

- No error

**Screenshots**

Screen Shot 2023-02-13 at 15 28 32

**Desktop (please complete the following information):**

- Version: v1.4.2
- Network: mainnet

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

Got `PublishError.Duplicate` errors at validator side even through it's a success. There are 2 reasons for it:
- when publishing a block using buider flow, builder may already published that block and gossip block reached our node (see #5140)
- when `vc` works with more than 1 beacon nodes, an item may reach one node first and spread across the network through gossipsub, when the other node calls `gossipsub.publish()` it throws `PublishError.Duplicate` error because it sees that item already through gossipsub (see #5098)

**Description**

- `ignoreDuplicatePublishError` option was added to gossipsub through https://github.com/ChainSafe/js-libp2p-gossipsub/pull/404/files and included in gossipsub `v6.2.0`
- Apply that to all `publish*()` APIs that come from `vc`

Closes #5140 #5098

### CODE REVIEW

**Review by g11tech — APPROVED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -4737,6 +4737,13 @@ bytes@3.1.2:
   resolved "https://registry.yarnpkg.com/bytes/-/bytes-3.1.2.tgz#8b0beeb98605adf1b128fa4386403c009e0221a5"
   integrity sha512-/Nf7TyzTx6S3yRJObOAV7956r8cr2+Oj8AC5dt8wSP3BQAoeX58NoHyCU8P8zGkNXStjTSi6fzO6F0pBdcYbEg==
 
+c-kzg@^1.0.9:
```
Comment (nflaig) [inline yarn.lock:4740]: is it intended that `c-kzg` is added in yarn.lock? I noticed it was showing up as a diff when running `yarn install` locally, this is fixed but this change seems unrelated to the PR

Comment (wemeetagain) [reply yarn.lock:4740]: yeah, its unrelated to this PR

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.6.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.6.0) :tada:
